### DISCUSSION TITLE
Do async afterEach calls resolve prior to completing?

### DISCUSSION THREAD

**Discussion by illegalnumbers** [vuejs/router#1825] [Help and Questions] [answered]
If I have an async function in an afterEach call does it resolve the promise before continuing on to the next route call? Looking at the following code compared to the beforeEach it does not seem so but I wanted to confirm:
[here](https://github.com/vuejs/router/blob/main/packages/router/src/router.ts#L910) vs [here2](https://github.com/vuejs/router/blob/7442d7f6b88d4da09ce42c22bc018377e8df0050/packages/router/src/router.ts#L1271)

**Comment by posva** [comment, marked as answer]
`afterEach()` happens after the navigation, so it's non blocking. `beforeEach()` annd `beforeResolve()` are the non blocking guards

**Reply by illegalnumbers** [reply]
Do you mean `beforeEach` and `beforeResolve` are the *blocking* guards?

**Reply by posva** [reply]
yes, the after are only hooks: https://router.vuejs.org/guide/advanced/navigation-guards.html#global-after-hooks
